rust/src/main.rs

```diff
@@ -65,6 +67,58 @@ async fn inner_async_main(args: Vec<String>) -> Result<i32> {
     .await?
 }
 
+/// Multicall entrypoint for `ostree-container`.
+async fn multicall_ostree_container(args: Vec<String>) -> Result<i32> {
+    ostree_ext::cli::run_from_iter(args).await?;
+    Ok(0)
+}
+
+/// Dispatch multicall binary to relevant logic, based on callname from `argv[0]`.
+async fn dispatch_multicall(callname: String, args: Vec<String>) -> Result<i32> {
+    match callname.as_str() {
+        // TODO(lucab): fix the final callname for this after ostree side at
+        // https://github.com/ostreedev/ostree/issues/2480 is settled.
+        "ostree-container" => multicall_ostree_container(args).await,
+        "rpm-ostree" | _ => inner_async_main(args).await,
+    }
+}
+
+/// Process a string from `argv[0]` into a clean callname.
+fn callname_from_argv0(argv0: &str) -> &str {
+    // SAFETY: `rsplit` iterator always contains at least one item.
+    let callname = argv0.rsplit('/').into_iter().nth(0).unwrap();
+    if callname.is_empty() {
+        "rpm-ostree"
+    } else {
+        callname
+    }
+}
+
+/// Gather arguments from command-line, and clean up `argv[0]` for multicall.
+fn gather_cli_args(argv: impl Iterator<Item = OsString>) -> Result<(String, Vec<String>)> {
```

Comment:
This doesn't matter at all here, but I think it's always more flexible to take `impl IntoIterator` if possible.  That's the first method we're calling on it below.  `into_iter()` on an `Iterator` is just a no-op: https://doc.rust-lang.org/src/core/iter/traits/collect.rs.html#238-246

Reply:
Oh yes, that signature was actually not intended, thanks for catching it.